tests/test_bs.py

```diff
@@ -0,0 +1,229 @@
+def setup_bootstrap(request):
+
+    def _setup_bootstrap_in_namespace(name_space):
+        global bs_info
+        bs_info = NodeInfo()
+        resp=create_deployment(BOOT_DEPLOYMENT_FILE, name_space)
+
+        bs_info.bs_deployment_name = resp.metadata._name
+        namespaced_pods = client.CoreV1Api().list_namespaced_pod(namespace=name_space).items
+        bootstrap_pod = next(filter(lambda i: i.metadata.name.startswith(bs_info.bs_deployment_name), namespaced_pods))
+        bs_info.bs_pod_name = bootstrap_pod.metadata.name
+
+        while True:
+            resp = client.CoreV1Api().read_namespaced_pod(name=bs_info.bs_pod_name, namespace=name_space)
+            if resp.status.phase != 'Pending':
+                break
+            time.sleep(1)
+
+        bs_info.bs_pod_ip = resp.status.pod_ip
+        bootstrap_pod_logs = client.CoreV1Api().read_namespaced_pod_log(name=bs_info.bs_pod_name, namespace=name_space)
+        match = re.search(r"Local node identity >> (?P<bootstarap_key>\w+)", bootstrap_pod_logs)
+        bs_info.bs_key = match.group('bootstarap_key')
+        return bs_info
+
+    def fin():
+        global bs_info
+        delete_deployment(bs_info.bs_deployment_name, testconfig['namespace'])
+
+    request.addfinalizer(fin)
+    return _setup_bootstrap_in_namespace(testconfig['namespace'])
+
+
+@pytest.fixture
+def setup_clients(request, setup_bootstrap):
+
+    def _setup_clients_in_namespace(name_space):
+        global bs_info, client_info
+        client_info = NodeInfo()
+        resp = create_deployment(CLIENT_DEPLOYMENT_FILE, name_space,
+                                 bootstrap_ip=bs_info.bs_pod_ip,
+                                 bootstrap_port='7513',
+                                 bootstrap_key=bs_info.bs_key)
+        client_info.bs_deployment_name = resp.metadata._name
+        namespaced_pods = client.CoreV1Api().list_namespaced_pod(namespace=name_space, include_uninitialized=True).items
+        client_pods = list(filter(lambda i: i.metadata.name.startswith(client_info.bs_deployment_name), namespaced_pods))
+
+        print("Number of client pods: {0}".format(len(client_pods)))
+        for c in client_pods:
+            while True:
+                resp = client.CoreV1Api().read_namespaced_pod(name=c.metadata.name, namespace=name_space)
+                if resp.status.phase != 'Pending':
+                    break
+                time.sleep(1)
+        return client_pods
+
+    def fin():
+        global client_info
+        delete_deployment(client_info.bs_deployment_name, testconfig['namespace'])
+
+    request.addfinalizer(fin)
+    return _setup_clients_in_namespace(testconfig['namespace'])
+
+
+#==============================================================================
+#    TESTS
+#==============================================================================
+
+current_index = 'kubernetes_cluster-2019.02.2*'
+
+
+def test_bootstrap(load_config, setup_bootstrap):
+    time.sleep(10)
```

Comment:
we do this only to wait for the record to show up in ES right? worth a comment.
also this might be useful in the future - https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html